README.md
# ProjectZeus
*****PLEASE DO NOT COPY OR DISTRIBUTE THIS PROJECT BUT YOU MAY FORK AND EDIT WITH MY PERMISSION*****

![PUBG Image](img/PUBG700x500.png)


This project that I will be working on when I have free time is a text-based, RPG type game with a PUBG twist.
In case you have no idea what PUBG (PlayerUnknowns Battlegrounds) is then I suggest you research first.

If anyone has any ideas that I can implement, feel free to contact me :)

Thanks,
OfficialMuffin




Changelog
***Start of Changelog***


src/test/java/com/projectzeus/AppTest.java
package com.projectzeus;

import junit.framework.Test;
import junit.framework.TestCase;
import junit.framework.TestSuite;

/**
 * Unit test for simple App.
 */
public class AppTest 
    extends TestCase
{
    /**
     * Create the test case
     *
     * @param testName name of the test case
     */
    public AppTest( String testName )
    {
        super( testName );
    }

    /**
     * @return the suite of tests being tested
     */
    public static Test suite()
    {
        return new TestSuite( AppTest.class );
    }

    /**
     * Rigourous Test :-)
     */
    public void testApp()
    {
        assertTrue( true );
    }
}


pubgWIP.java
package WIP;

import java.util.Random;
import java.util.Scanner;

public class pubgWIP {
    public static void main(String[] args) {
        //System Objects
        Scanner in = new Scanner(System.in);
        Random rand = new Random();

        //Game and Enemy Variables
        String[] enemies = { "Enemy with Assault Rifle", "Enemy with SMG", "Enemy with Sniper", "Enemy with Semi Auto Sniper"};
        //Find a way to connect attackDamage with the appropriate enemy type
        //Integer[] attackDamage = { 25, 15, 35, 45 };
          int enemyAttackDamage = 25;
          int maxEnemyHealth = 100;
//        final int assaultAttackDamage = 25;
//        final int smgAttackDamage = 15;
//        final int sniperAttackDamage = 35;
//        final int semiAutoSniperAttackDamage = 45;

        //Player Variables
        int health = 100;
        int playerAttackDamage = 25;
        int numHealthBandages = 0;
        int numMedkits = 0;
        int healAmount = 25;
        int bandageDropChance = 35; //Percentage
        int medkitDropChance = 5; //Percentage
        int medkitHealAmount = 75; //Percentage


        //Variable for running game
        boolean running = true;

        System.out.println("Welcome to the Battleground!");

        GAME:
        while (running) {
            System.out.println("-------------------------------------------------------------------");

            int enemyHealth = rand.nextInt(maxEnemyHealth);
            String enemy = enemies[rand.nextInt(enemies.length)];
            System.out.println("\t# " + enemy + " spotted! #\n");

            while (enemyHealth > 0) { //While enemy is alive
                System.out.println("\tYour HP: " + health);
                System.out.println("\t" + enemy + " HP: " + enemyHealth);
                System.out.println("\n\tWhat would you like to do?");
                System.out.println("\t1. Attack");
                System.out.println("\t2. Use Bandage");
                System.out.println("\t3. Use Medkit");
                System.out.println("\t4. Keep Running");

                String input = in.nextLine();
                if(input.equals("1")) {
                    int damageDealt = rand.nextInt(playerAttackDamage);
                    int damageTaken = rand.nextInt(enemyAttackDamage);

                    enemyHealth -= damageDealt;
                    health -= damageTaken;

                    System.out.println("\t> You hit " + enemy + " for " + damageDealt + " damage.");
                    System.out.println("\t> You take " + damageTaken + " damage!");

                    if(health < 1) {
                        System.out.println("*****You have been killed by " + enemy + "*****");
                        break;
                    }

                } //Healing Information
                else if(input.equals("2")) {
                    if(numHealthBandages > 0) {
                        health += healAmount;
                        numHealthBandages--;
                        System.out.println("\t Using bandages...");
                        //Thread.sleep(5000);
                        System.out.println("You have healed for " + healAmount + "."
                                + "\n\t> You now have "+ health + " HP."
                                + "\n\t> You have " + numHealthBandages + "bandages left.\n");

                    }
                    else {
                        System.out.println("\t> You have no bandages left! Defeat enemies to pick up more!");
                    }

                }
                else if (input.equals("3")) {
                    if(numMedkits > 0) {
                        //Adjust medkit heal amount
                        health += medkitHealAmount;
                        numMedkits--;
                        System.out.println("\tUsing Medkit...");
                        //Thread.sleep(7500);
                        System.out.println("You have healed for " + medkitHealAmount + "."
                                + "\n\t> You now have " + health + " HP."
                                + "\n\t> You have " + numMedkits + " medkits left.\n");
                    }
                    else {
                        System.out.println("\t> You have no medkits left! Defeat enemies to pick up more!");
                    }


                }
                else if(input.equals("4")) {
                    System.out.println("You run away from the " + enemy + "!");
                    continue GAME;
                }
                else {
                    System.out.println("\tInvalid Command");
                }
            }
            if (health < 1) {
                System.out.println("*****You didn't get the Chicken Dinner :(*****");
                break;
            }
            //Useful Information
            System.out.println("-------------------------------------------------------------------");
            System.out.println(" # " + enemy + " was killed! # ");
            System.out.println(" # You have " + health + " HP left! #");
            if (rand.nextInt(100) > bandageDropChance) {
                numHealthBandages++;
                System.out.println(" # The " + enemy + " had " + numHealthBandages + " bandage(s)! # ");
                System.out.println(" # You now have " + numHealthBandages + " bandage(s). # ");
            }
            if (rand.nextInt(100) > medkitDropChance){
                numMedkits++;
                System.out.println(" # The " + enemy + " had " + numMedkits + " medkit(s)! # ");
                System.out.println(" # You now have " + numMedkits + " medkit(s). # ");
            }
            System.out.println("-------------------------------------------------------------------");
            System.out.println("What would you like to do now?");
            System.out.println("1. Continue Fighting");
            System.out.println("2. Exit area");

            String input = in.nextLine();

            while (!input.equals("1") && !input.equals("2")) {
                System.out.println("Invalid Command");
                input = in.nextLine();
            }
            if (input.equals("1")) {
                System.out.println("You continue on!");
            }
            else if (input.equals("2")) {
                System.out.println("You exit the battlegrounds!");
                break;
            }

        }

        System.out.println("#####################################");
        System.out.println("#         THANKS FOR PLAYING!       #");
        System.out.println("#####################################");

    }
}



src/main/java/com/projectzeus/enemy/SniperEnemy.java
package com.projectzeus.enemy;

import java.util.Random;

public class SniperEnemy implements Enemy {
    private final String name = "Enemy with Sniper";
    private final int damageModifier = 35;
    private final int maxHealth = 100;

    private int health;
    
    private Random rand;

    public SniperEnemy() {
	rand = new Random();

	this.health = rand.nextInt(maxHealth);
    }

    public int getHealth() {
	return this.health;
    }

    public void dealDamage(int damage) {
	this.health -= damage;

	if (this.health < 0) {
	    this.health = 0;
	}
    }
    
    // Get the enemy name
    public String getName() {
	return this.name;
    }

    // Calculate the attack damage for the given class of enemy
    public int getAttackDamage() {
	return rand.nextInt(damageModifier);
    }
}


src/main/java/com/projectzeus/enemy/AssaultRifleEnemy.java
package com.projectzeus.enemy;

import java.util.Random;

public class AssaultRifleEnemy implements Enemy {
    private final String name = "Enemy with Assault Rifle";
    private final int damageModifier = 25;
    private final int maxHealth = 100;

    private int health;
    
    private Random rand;

    public AssaultRifleEnemy() {
	rand = new Random();

	this.health = rand.nextInt(maxHealth);
    }

    public int getHealth() {
	return this.health;
    }

    public void dealDamage(int damage) {
	this.health -= damage;

	if (this.health < 0) {
	    this.health = 0;
	}
    }
    
    // Get the enemy name
    public String getName() {
	return this.name;
    }

    // Calculate the attack damage for the given class of enemy
    public int getAttackDamage() {
	return rand.nextInt(damageModifier);
    }
}


src/main/java/com/projectzeus/enemy/SemiAutoSniperEnemy.java
package com.projectzeus.enemy;

import java.util.Random;

public class SemiAutoSniperEnemy implements Enemy {
    private final String name = "Enemy with Semi Auto Sniper";
    private final int damageModifier = 45;
    private final int maxHealth = 100;

    private int health;
    
    private Random rand;

    public SemiAutoSniperEnemy() {
	rand = new Random();

	this.health = rand.nextInt(maxHealth);
    }

    public int getHealth() {
	return this.health;
    }

    public void dealDamage(int damage) {
	this.health -= damage;

	if (this.health < 0) {
	    this.health = 0;
	}
    }
    
    // Get the enemy name
    public String getName() {
	return this.name;
    }

    // Calculate the attack damage for the given class of enemy
    public int getAttackDamage() {
	return rand.nextInt(damageModifier);
    }
}


src/main/java/com/projectzeus/App.java
package com.projectzeus;

import java.util.Random;
import java.util.Scanner;

import com.projectzeus.enemy.Enemy;
import com.projectzeus.enemy.EnemyFactory;

public class App {
    public static void main(String[] args) {
        //System Objects
        Scanner in = new Scanner(System.in);
        Random rand = new Random();

        //Player Variables
        int health = 100;
        int playerAttackDamage = 25;
        int numHealthBandages = 0;
        int numMedkits = 0;
        int healAmount = 25;
        int bandageDropChance = 35; //Percentage
        int medkitDropChance = 5; //Percentage
        int medkitHealAmount = 75; //Percentage
        int numPainkillers = 0;
        int painkillerDropChance = 15;
        int painkillerHealAmount = 10;
        int numEnergyDrinks = 0;
        int energyDrinkDropChance = 20;
        int energyDrinkHealAmount = 5;

        //Variable for running game
        boolean running = true;

        System.out.println("Welcome to PlayerUnknown's Battlegrounds Text-Based RPG!");

        GAME:
        while (running) {
            System.out.println("-------------------------------------------------------------------");

	    Enemy enemy = EnemyFactory.getEnemy();
            System.out.println("\t# " + enemy.getName() + " spotted! #\n");

            while (enemy.getHealth() > 0) { //While enemy is alive
                System.out.println("\tYour HP: " + health);
                System.out.println("\t" + enemy.getName() + " HP: " + enemy.getHealth());
                System.out.println("\n\tWhat would you like to do?");
                System.out.println("\t1. Attack");
                System.out.println("\t2. Use Bandage");
                System.out.println("\t3. Use Energy drink");
                System.out.println("\t4. Use Painkillers");
                System.out.println("\t5. Use Medkit");
                System.out.println("\t6. Keep Running");

                String input = in.nextLine();
                if(input.equals("1")) {
                    int damageDealt = rand.nextInt(playerAttackDamage);
                    int damageTaken = enemy.getAttackDamage();

		    enemy.dealDamage(damageDealt);
                    health -= damageTaken;

                    System.out.println("\t> You hit " + enemy.getName() + " for " + damageDealt + " damage.");
                    System.out.println("\t> You take " + damageTaken + " damage!");

                    if(health < 1) {
                        System.out.println("*****You have been killed by " + enemy.getName() + "*****");
                        break;
                    }

                } //Healing Information
                else if(input.equals("2")) {
                    if(numHealthBandages > 0) {
                        health += healAmount;
                        numHealthBandages--;
                        System.out.println("\t Using bandages...");
                        //Thread.sleep(5000);
                        System.out.println("You have healed for " + healAmount + "."
                                + "\n\t> You now have "+ health + " HP."
                                + "\n\t> You have " + numHealthBandages + "bandages left.\n");

                    }
                    else {
                        System.out.println("\t> You have no bandages left! Defeat enemies to pick up more!");
                    }

                }
                else if (input.equals("3")) {
                    if(numEnergyDrinks > 0) {
                        health += energyDrinkHealAmount;
                        numEnergyDrinks--;
                        System.out.println("\tUsing Energy Drink...");
                        //Thread.sleep(2500);
                        System.out.println("You have healed for " + energyDrinkHealAmount + "."
                                + "\n\t> You now have " + health + " HP."
                                + "\n\t> You have " + numEnergyDrinks + " energy drinks left.\n");
                    }
                    else {
                        System.out.println("\t> You have no energy drinks left! Defeat enemies to pick up more!");
                    }
                }
                else if (input.equals("4")) {
                    if(numPainkillers > 0) {
                        health += painkillerHealAmount;
                        numPainkillers--;
                        System.out.println("\tUsing Painkillers...");
                        //Thread.sleep(3500);
                        System.out.println("You have healed for " + painkillerHealAmount + "."
                                + "\n\t> You now have " + health + " HP."
                                + "\n\t> You have " + numPainkillers + " painkillers left.\n");
                    }
                    else {
                        System.out.println("\t> You have no painkillers left! Defeat enemies to pick up more!");
                    }
                }
                else if (input.equals("5")) {
                    if(numMedkits > 0) {
                        //Adjust medkit heal amount
                        health += medkitHealAmount;
                        numMedkits--;
                        System.out.println("\tUsing Medkit...");
                        //Thread.sleep(7500);
                        System.out.println("You have healed for " + medkitHealAmount + "."
                                + "\n\t> You now have " + health + " HP."
                                + "\n\t> You have " + numMedkits + " medkits left.\n");
                    }
                    else {
                        System.out.println("\t> You have no medkits left! Defeat enemies to pick up more!");
                    }
                }
                else if(input.equals("4")) {
                    System.out.println("You run away from the " + enemy.getName() + "!");
                    continue GAME;
                }
                else {
                    System.out.println("\tInvalid Command");
                }
            }
            if (health < 1) {
                System.out.println("*****You didn't get the Chicken Dinner :(*****");
                break;
            }
            //Useful Information
            System.out.println("-------------------------------------------------------------------");
            System.out.println(" # " + enemy.getName() + " was killed! # ");
            System.out.println(" # You have " + health + " HP left! #");
            if (rand.nextInt(100) > bandageDropChance) {
                numHealthBandages++;
                System.out.println(" # The " + enemy.getName() + " had " + numHealthBandages + " bandage(s)! # ");
                System.out.println(" # You now have " + numHealthBandages + " bandage(s). # ");
            }
            if (rand.nextInt(100) > medkitDropChance){
                numMedkits++;
                System.out.println(" # The " + enemy.getName() + " had " + numMedkits + " medkit(s)! # ");
                System.out.println(" # You now have " + numMedkits + " medkit(s). # ");
            }
	    if (rand.nextInt(100) > energyDrinkDropChance) {
                numEnergyDrinks++;
                System.out.println(" # The " + enemy.getName() + " had " + numEnergyDrinks + " energy drink(s)! # ");
                System.out.println(" # You now have " + numEnergyDrinks + " energy drink(s). # ");
            }
            if (rand.nextInt(100) > painkillerDropChance) {
                numPainkillers++;
                System.out.println(" # The " + enemy.getName() + " had " + numPainkillers + " painkiller(s)! # ");
                System.out.println(" # You now have " + numPainkillers + " painkiller(s). # ");
            } 
            System.out.println("-------------------------------------------------------------------");
            System.out.println("What would you like to do now?");
            System.out.println("1. Continue Fighting");
            System.out.println("2. Exit area");

            String input = in.nextLine();

            while (!input.equals("1") && !input.equals("2")) {
                System.out.println("Invalid Command");
                input = in.nextLine();
            }
            if (input.equals("1")) {
                System.out.println("You continue on!");
            }
            else if (input.equals("2")) {
                System.out.println("You exit the battlegrounds!");
                break;
            }

        }

        System.out.println("#####################################");
        System.out.println("#         THANKS FOR PLAYING!       #");
        System.out.println("#####################################");

    }
}

